v2/devdoc/log_sink_etw_requirements.md

```diff
@@ -0,0 +1,179 @@
+# `log_sink_etw` requirements
+
+`log_sink_etw` implements a log sink interface that logs the events as ETW events.
+
+## References
+
+[Event Tracing For Windows](https://learn.microsoft.com/en-us/windows-hardware/drivers/devtest/event-tracing-for-windows--etw-)
+
+## Exposed API
+
+```
+    extern const LOG_SINK_IF log_sink_etw;
+```
+
+### log_sink_etw.log_sink_log
+
+The signature of `log_sink_etw.log_sink_log` is:
+
+```c
+typedef void (*LOG_SINK_LOG_FUNC)(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* file, const char* func, int line, const char* message_format, ...);
+```
+
+`log_sink_etw.log_sink_log` implements logging to an ETW provider.
+
+If `message_format` is `NULL`, `log_sink_etw.log_sink_log` shall fail and return.
+
+`log_sink_etw_log` shall register the ETW TraceLogging provider by calling `TraceLoggingRegister` if it was not already done.
+
+`log_sink_etw_log` shall use as provider GUID `DAD29F36-0A48-4DEF-9D50-8EF9036B92B4`.
+
+Note this can (and should) be improved to be configurable later.
```

Comment:
ha! I was going to ask about this since it would be nice to have a different provider for ELS and be able to easily filter between EBS and ELS traces (and to not confuse our PubSub friends who are becoming experts in analyzing our traces...)

Reply:
Yes, I think it is going to be a separate task. I would also like to have different providers.